verifier/rest/rest_test.go

```diff
@@ -194,3 +196,222 @@ func TestConvertTDXProtoToREST(t *testing.T) {
+						AllowedPrincipalTags: &ccpb.TokenOptions_AwsPrincipalTagsOptions_AllowedPrincipalTags{},
+					},
+				},
+			},
+		},
+		{
+			name: "MissingAudNonceTokenType",
+			tokenOptions: &models.TokenOptions{
+				PrincipalTagOptions: &models.AWSPrincipalTagsOptions{
+					AllowedPrincipalTags: &models.AllowedPrincipalTags{},
+				},
+			},
+			wantpb: &ccpb.TokenOptions{
+				TokenTypeOptions: &ccpb.TokenOptions_AwsPrincipalTagsOptions_{
+					AwsPrincipalTagsOptions: &ccpb.TokenOptions_AwsPrincipalTagsOptions{
+						AllowedPrincipalTags: &ccpb.TokenOptions_AwsPrincipalTagsOptions_AllowedPrincipalTags{},
+					},
+				},
+			},
+		},
+		{
+			name: "TokenOptionsHappyPath",
+			tokenOptions: &models.TokenOptions{
+				Audience:  "TestingAudience",
+				Nonces:    []string{"thisisthefirstnonce", "thisisthesecondnonce"},
+				TokenType: "AWS_PRINCIPALTAGS",
+				PrincipalTagOptions: &models.AWSPrincipalTagsOptions{
+					AllowedPrincipalTags: &models.AllowedPrincipalTags{
+						ContainerImageSignatures: &models.ContainerImageSignatures{
+							KeyIDs: []string{"abcdefg", "12345"},
+						},
+					},
+				},
+			},
+			wantpb: &ccpb.TokenOptions{
+				Audience:  "TestingAudience",
+				Nonce:     []string{"thisisthefirstnonce", "thisisthesecondnonce"},
+				TokenType: ccpb.TokenType_TOKEN_TYPE_AWS_PRINCIPALTAGS,
+				TokenTypeOptions: &ccpb.TokenOptions_AwsPrincipalTagsOptions_{
+					AwsPrincipalTagsOptions: &ccpb.TokenOptions_AwsPrincipalTagsOptions{
+						AllowedPrincipalTags: &ccpb.TokenOptions_AwsPrincipalTagsOptions_AllowedPrincipalTags{
+							ContainerImageSignatures: &ccpb.TokenOptions_AwsPrincipalTagsOptions_AllowedPrincipalTags_ContainerImageSignatures{
+								KeyIds: []string{"abcdefg", "12345"},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "OIDCTokenType",
+			tokenOptions: &models.TokenOptions{
+				TokenType: "OIDC",
+			},
+			wantpb: &ccpb.TokenOptions{
+				TokenType: ccpb.TokenType_TOKEN_TYPE_OIDC,
+			},
+		},
+		{
+			name: "OIDCTokenType",
+			tokenOptions: &models.TokenOptions{
+				TokenType: "OIDC",
+			},
+			wantpb: &ccpb.TokenOptions{
+				TokenType: ccpb.TokenType_TOKEN_TYPE_OIDC,
+			},
+		},
+		{
+			name: "LimitedAWSTokenType",
+			tokenOptions: &models.TokenOptions{
+				TokenType: "LIMITED_AWS",
+			},
+			wantpb: &ccpb.TokenOptions{
+				TokenType: ccpb.TokenType_TOKEN_TYPE_LIMITED_AWS,
+			},
+		},
+		{
+			name: "SingleNonce",
+			tokenOptions: &models.TokenOptions{
+				Nonces: []string{"thisistheonlynonce"},
+			},
+			wantpb: &ccpb.TokenOptions{
+				Nonce: []string{"thisistheonlynonce"},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		pbTokenOpts := convertTokenOptionsToREST(tc.tokenOptions)
+		ok, result := compareTokenOptionsPbs(tc.wantpb, pbTokenOpts)
+		if !ok {
+			t.Errorf("%v: %s", tc.name, result)
+		}
+	}
+}
+
+func compareTokenOptionsPbs(want *ccpb.TokenOptions, got *ccpb.TokenOptions) (bool, string) {
```

Comment:
Can we do cmp.Equal on the structs, so we don't need to have a customized comparator.